test/functional/wallet_multisig_descriptor_psbt.py

```diff
@@ -0,0 +1,158 @@
+)
+
+
+class WalletMultisigDescriptorPSBTTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 3
+        self.setup_clean_chain = True
+        self.wallet_names = []
+        self.extra_args = [["-keypool=100"]] * self.num_nodes
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+        self.skip_if_no_sqlite()
+
+    def _get_xpub(self, wallet):
+        """Extract the wallet's xpubs using `listdescriptors` and pick the one from the `pkh` descriptor since it's least likely to be accidentally reused (legacy addresses)."""
+        descriptor = next(filter(lambda d: d["desc"].startswith("pkh"), wallet.listdescriptors()["descriptors"]))
+        return descriptor["desc"].split("]")[-1].split("/")[0]
+
+    def _check_psbt(self, psbt, to, value, multisig):
+        """Helper method for any of the N participants to check the psbt with decodepsbt and verify it is OK before signing."""
+        tx = multisig.decodepsbt(psbt)["tx"]
+        amount = 0
+        for vout in tx["vout"]:
+            address = vout["scriptPubKey"]["address"]
+            assert_equal(multisig.getaddressinfo(address)["ischange"], address != to)
+            if address == to:
+                amount += vout["value"]
+        assert_approx(amount, float(value), vspan=0.001)
+
+    def generate_and_exchange_xpubs(self, participants):
+        """Every participant generates an xpub. The most straightforward way is to create a new descriptor wallet. Avoid reusing this wallet for any other purpose.."""
+        for i, node in enumerate(participants):
+            node.createwallet(wallet_name=f"participant_{i}", descriptors=True)
+            yield self._get_xpub(node.get_wallet_rpc(f"participant_{i}"))
+
+    def participants_import_descriptors(self, participants, xpubs):
+        """The multisig is created by importing the following descriptors. The resulting wallet is watch-only and every participant can do this."""
+        # some simple validation
+        assert_equal(len(xpubs), self.N)
+        # a sanity-check/assertion, this will throw if the base58 checksum of any of the provided xpubs are invalid
+        for xpub in xpubs:
+            base58_to_byte(xpub)
+
+        for i, node in enumerate(participants):
+            node.createwallet(wallet_name=f"{self.name}_{i}", blank=True, descriptors=True, disable_private_keys=True)
+            multisig = node.get_wallet_rpc(f"{self.name}_{i}")
+            result = multisig.importdescriptors([
+                {  # receiving addresses (internal: False)
+                    "desc": descsum_create(f"wsh(sortedmulti({self.M},{f'/{0}/*,'.join(xpubs)}/{0}/*))"),
+                    "active": True,
+                    "internal": False,
+                    "timestamp": "now",
+                },
+                {  # change addresses (internal: True)
+                    "desc": descsum_create(f"wsh(sortedmulti({self.M},{f'/{1}/*,'.join(xpubs)}/{1}/*))"),
```

Comment:
nit: By using `importdescriptors`, it eliminates the need for `descsum_create`, so the test will only use node functions without an additional script for the checksum.

`from test_framework.descriptors import descsum_create` can also be removed.

```suggestion
            multisig = node.get_wallet_rpc(f"{self.name}_{i}")
            
            external_desc = multisig.getdescriptorinfo(f"wsh(sortedmulti({self.M},{f'/{0}/*,'.join(xpubs)}/{0}/*))")
            internal_desc = multisig.getdescriptorinfo(f"wsh(sortedmulti({self.M},{f'/{1}/*,'.join(xpubs)}/{1}/*))")
            
            result = multisig.importdescriptors([
                {  # receiving addresses (internal: False)
                    "desc": external_desc["descriptor"],
                    "active": True,
                    "internal": False,
                    "timestamp": "now",
                },
                {  # change addresses (internal: True)
                    "desc": internal_desc["descriptor"],
```

Reply:
I took this suggestion and added the improvement in https://github.com/bitcoin/bitcoin/pull/22067/commits/1f20501efce041d34e63ab9a11359bedf4a82cd5. Marking this resolved